### DISCUSSION TITLE
OnModelChanged not executed

### DISCUSSION THREAD

**Discussion by weltesche** [MarimerLLC/csla#4222] [Questions] [answered]
Using .NET 8 / Csla 8.2.6 / Wpf, Xaml

In Csla 8.2.6 OnModelChanged is not executed when changing and saving my BO.

Step 1: When initially fetching the BO it returns with IsDirty false (as expected) and OnModelChanged in Csla.Xaml.ViewModelBase is executed. OK.

Step 2: When changing and saving the initially fetched BO, OnModelChanged is NOT executed and the BO returns with IsDirty true (which should be false).  Although the changed state has been stored in the db.

Step 3: Also, when again changing the saved BO and canceling the changes, the BO shows the initially fetched state from Step 1 (and not the saved state from Step 2 as expected).

Csla 6.2.2 does not show this behavior, everything works as expected.

Maybe I missed a config in Csla 8.2.6?

Best regards

**Comment by StefanOssendorf** [comment, marked as answer]
Hi. That is intended behavior. Instead of re-assigning the model property.
The changes coming from the server are merged into the current instance of
the model. You can see that here:
https://github.com/MarimerLLC/csla/blob/fed03f30c2207d9991087a67defe6c9ce004568c/Source/Csla.Xaml.Shared/ViewModelBase.cs#L585

Hth

Grimbart ***@***.***> schrieb am So., 15. Sept. 2024, 14:15:

**Comment by weltesche** [comment]
Thanks for the quick response.
I really missed the SaveAndMerge feature.

**Reply by Chicagoan2016** [reply]
Out of curiosity, how are you going to solve it?

**Reply by weltesche** [reply]
I will probably refetch my BO after each save command.
Or maybe I override the DoSaveAsync() in Csla.Xaml.ViewModelBase, commenting out the GraphMerger.MergeGraph.

**Reply by rockfordlhotka** [reply]
You could maybe override `DoSaveAsync`, call its base to do the work, then call `OnModelChanged`?
